packages/base-ui/src/views/app.tsx

```diff
@@ -96,6 +96,12 @@ export function App(props: IUniverAppProps) {
 
     const [locale, setLocale] = useState<LocaleType>(localeService.getLocale().getCurrentLocale());
 
+    useEffect(() => {
+        localeService.getLocale().locale$.subscribe((locale) => {
```

Comment:
Be careful when you subscribe to an observable. Make sure you will unsubscribe especially ui is upper level to core.